src/_pytest/recwarn.py

```diff
@@ -139,6 +139,11 @@ def warns(
           ...
         Failed: DID NOT WARN. No warnings of type ...UserWarning... were emitted...
 
+    **Using with** ``pytest.mark.parametrize``
+
+    When using :ref:`pytest.mark.parametrize ref` it is possible to parametrize tests
+    such that some runs raise a warning and others do not.
```

Comment:
Could you please add an example here to illustrate this?

Reply:
could we put a link to [Parametrizing conditional raising](https://docs.pytest.org/en/7.1.x/example/parametrize.html#parametrizing-conditional-raising) and point out that it is done in a similar way?

Reply:
For example: 
```
This could be achieved in the same way as with exceptions, see 
:ref:`parametrizing_conditional_raising` for an example.
```

Reply:
@nicoddemus we have added the link. Please let us know if this is acceptable or if a specific example is needed.